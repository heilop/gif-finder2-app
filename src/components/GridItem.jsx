import { PropTypes } from 'prop-types'

export const GridItem = ({title, url, id}) => {
  return (
    <div className="card">
      <img src={ url } alt={ title } />
      <p>{ title }</p>
    </div>
  )
}

// set props types
GridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
