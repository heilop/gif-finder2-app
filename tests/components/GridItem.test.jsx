import { render, screen } from '@testing-library/react'
import { GridItem } from '../../src/components/GridItem'

describe('Test on <GridItem/>', () => {

  const title = 'Demo title';
  const url = 'https://some-url/demo.jpg';

  test('should match with snapshot', () => {
    const { container } = render (<GridItem title={ title } url={ url } />)
    expect(container).toMatchSnapshot()
  })

  test('should have image with URL and ALT attributes', () => {
    render (<GridItem title={ title } url={ url } />)
    // screen.debug();
    // expect( screen.getByRole('img').src ).toBe(url);
    // expect( screen.getByRole('img').alt ).toBe(title);
    const { src, alt } = screen.getByRole('img');
    expect( src ).toBe(url);
    expect( alt ).toBe(title);
  })

  test('should shown title of component', () => {
    render(<GridItem title={ title } url={ url } />);
    expect( screen.getByText(title) ).toBeTruthy();
   })
 })
