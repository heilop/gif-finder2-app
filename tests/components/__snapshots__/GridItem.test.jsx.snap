// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GridItem/> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Demo title"
      src="https://some-url/demo.jpg"
    />
    <p>
      Demo title
    </p>
  </div>
</div>
`;
